optimade/server/utils.py

```diff
@@ -113,16 +113,75 @@ def handle_response_fields(
     return new_results
 
 
+def get_included_relationships(
+    results: Union[EntryResource, List[EntryResource]],
+    entry_collections: Dict[str, EntryCollection],
+) -> Dict[str, List[EntryResource]]:
+    """Filters the included relationships and makes the appropriate compound request
+    to include them in the response.
+
+    Parameters:
+        results: list of returned documents.
+        entry_collections: dictionary containing collections to query, with key
+            based on endpoint type.
+
+    Returns:
+        Dictionary with the same keys as entry_collections, each containing the list
+            of resource objects for that entry type.
+
+    """
+    from collections import defaultdict
+
+    if not isinstance(results, list):
+        results = [results]
+
+    endpoint_includes = defaultdict(dict)
+    for doc in results:
+        # convert list of references into dict by ID to only included unique IDs
+        if doc is None:
+            continue
+
+        relationships = doc.relationships
+        if relationships is None:
+            continue
+
+        relationships = relationships.dict()
+        for entry_type in entry_collections:
+            entry_relationship = relationships.get(entry_type, {})
+            if entry_relationship is not None:
+                refs = entry_relationship.get("data", [])
+                for ref in refs:
+                    # could check here and raise a warning if any IDs clash
+                    endpoint_includes[entry_type][ref["id"]] = ref
+
+    included = {}
+    for entry_type in endpoint_includes:
+        compound_filter = " AND ".join(
```

Comment:
Shouldn't this be OR? Since otherwise we are looking for related entries that have `id="x" AND id="y"`, which should hopefully never return any results ;)